app/admin/users-list.component.ts

```diff
@@ -0,0 +1,60 @@
+﻿import { Component, OnInit } from '@angular/core';
+import { CmsApiService } from '../shared/api/cms-api.service';
+import { Observable } from 'rxjs';
+import { UserService } from '../shared/user/user.service';
+import { User } from '../shared/user/user.model';
+import { UsersFilter } from './filter.pipe';
+import { UsersSorter } from './sort.pipe';
+import { PaginatePipe, PaginationService, PaginationControlsCmp } from 'ng2-pagination';
+
+@Component({
+    selector: 'hip-users-list',
+    templateUrl: '../app/admin/users-list.component.html',
+    providers: [UserService, PaginationService, CmsApiService],
+    directives: [PaginationControlsCmp],
+    pipes: [UsersFilter, UsersSorter, PaginatePipe]
+})
+
+export class UsersListComponent implements OnInit {
+
+    errorMessage: any;
+    query: string = '';
+    key: string = '';
+    direction: number = -1;
+    roles = ['Student', 'Supervisor', 'Administrator'];
+    options = ['Last Name', 'First Name', 'Email', 'Role'];
+
```

Comment:
Maybe make 'Email' as default options, since email is a required field on registration